app/lib/frontend/handlers/package.dart

```diff
@@ -324,61 +325,77 @@ Future<shelf.Response> packageActivityLogHandler(
 
 @visibleForTesting
 Future<PackagePageData> loadPackagePageData(
-  String packageName,
+  Package package,
   String? versionName,
   String? assetKind,
 ) async {
-  final package = await packageBackend.lookupPackage(packageName);
-  if (package == null || package.isNotVisible) {
-    // note: this should not happen, as isVisible check is done before calling the method
-    throw NotFoundException.resource('package "$package"');
-  }
+  final packageName = package.name!;
+  versionName ??= package.latestVersion;
+
+  final latestReleasesFuture =
+      Future(() => packageBackend.latestReleases(package));
+
+  final isLikedFuture = Future(
+    () async => requestContext.isNotAuthenticated
+        ? false
+        : await likeBackend.getPackageLikeStatus(
+                requestContext.authenticatedUserId!, package.name!) !=
+            null,
+  );
+
+  final selectedVersionFuture = Future(
```

Comment:
This future may go unawaited! It's possible that `getScoreCardData` won't await it!

Reply:
In general the fact that we have to trace it that way suggests that we need to do this differently.